pkg/controller/backup_deletion_controller.go

```diff
@@ -54,250 +49,206 @@ import (
-		backupClient:              backupClient,
-		restoreLister:             restoreLister,
-		restoreClient:             restoreClient,
-		backupTracker:             backupTracker,
-		resticMgr:                 resticMgr,
-		podvolumeBackupLister:     podvolumeBackupLister,
-		kbClient:                  kbClient,
-		snapshotLocationLister:    snapshotLocationLister,
-		csiSnapshotLister:         csiSnapshotLister,
-		csiSnapshotContentLister:  csiSnapshotContentLister,
-		csiSnapshotClient:         csiSnapshotClient,
-		metrics:                   metrics,
-		helper:                    helper,
-		// use variables to refer to these functions so they can be
-		// replaced with fakes for testing.
-		newPluginManager:  newPluginManager,
-		backupStoreGetter: backupStoreGetter,
-
-		clock: &clock.RealClock{},
-	}
-
-	c.syncHandler = c.processQueueItem
-	c.processRequestFunc = c.processRequest
-
-	deleteBackupRequestInformer.Informer().AddEventHandler(
-		cache.ResourceEventHandlerFuncs{
-			AddFunc: c.enqueue,
-		},
-	)
-
-	c.resyncPeriod = time.Hour
-	c.resyncFunc = c.deleteExpiredRequests
-
-	return c
-}
-
-func (c *backupDeletionController) processQueueItem(key string) error {
-	log := c.logger.WithField("key", key)
-	log.Debug("Running processItem")
-
-	ns, name, err := cache.SplitMetaNamespaceKey(key)
-	if err != nil {
-		return errors.Wrap(err, "error splitting queue key")
-	}
-
-	req, err := c.deleteBackupRequestLister.DeleteBackupRequests(ns).Get(name)
-	if apierrors.IsNotFound(err) {
-		log.Debug("Unable to find DeleteBackupRequest")
-		return nil
-	}
-	if err != nil {
-		return errors.Wrap(err, "error getting DeleteBackupRequest")
-	}
-
-	switch req.Status.Phase {
-	case velerov1api.DeleteBackupRequestPhaseProcessed:
-		// Don't do anything because it's already been processed
-	default:
-		// Don't mutate the shared cache
-		reqCopy := req.DeepCopy()
-		return c.processRequestFunc(reqCopy)
+	newPluginManager func(logrus.FieldLogger) clientmgmt.Manager,
+	backupStoreGetter persistence.ObjectBackupStoreGetter,
+	podvolumeBackupLister velerov1listers.PodVolumeBackupLister,
+) *backupDeletionReconciler {
+	return &backupDeletionReconciler{
+		Client:                client,
+		logger:                logger,
+		backupTracker:         backupTracker,
+		resticMgr:             resticMgr,
+		metrics:               metrics,
+		clock:                 clock.RealClock{},
+		discoveryHelper:       helper,
+		newPluginManager:      newPluginManager,
+		backupStoreGetter:     backupStoreGetter,
+		podvolumeBackupLister: podvolumeBackupLister,
 	}
+}
 
-	return nil
+func (r *backupDeletionReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// Make sure the expired requests can be deleted eventually
+	s := kube.NewPeriodicalEnqueueSource(r.logger, mgr.GetClient(), &velerov1api.DeleteBackupRequestList{}, time.Hour)
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&velerov1api.DeleteBackupRequest{}).
+		Watches(s, nil).
```

Comment:
Looks like the controller will reconcile the BackupDelete request by one hour periodically.
If I understand it correct, this shouldn't be the expected behavior to handle BackupDelete.
BackupDelete should be reconciled when its created, not periodically.

Reply:
This ensures the controller handle the new request immediately and handle the old requests periodically.